rosbag2_transport/test/rosbag2_transport/test_record_all_include_unpublished_topics.cpp

```diff
@@ -81,13 +80,12 @@ TEST_F(RecordIntegrationTestFixture, record_all_include_unpublished_true_include
   MockSequentialWriter & mock_writer =
     static_cast<MockSequentialWriter &>(writer.get_implementation_handle());
 
-  size_t expected_messages = 0;
   rosbag2_test_common::wait_until_shutdown(
     std::chrono::seconds(2),
-    [&mock_writer, &expected_messages]() {
-      return mock_writer.get_messages().size() > expected_messages;
+    [&mock_writer]() {
+      // Return false so we wait the full two seconds
+      return false;
     });
```

Comment:
@theseankelly nit: looking at this code, I suspect that what you actually want is simply `rclcpp::sleep_for(std::chrono::seconds(2))`. Same elsewhere.

Reply:
indeed. thanks for the simplication.